src/R/Components/Impl/Plots/Implementation/Commands/PlotDevicePasteCommand.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using System.Windows;
+using Microsoft.R.Components.Controller;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.R.Components.Plots.ViewModel;
+using Microsoft.R.Host.Client;
+
+namespace Microsoft.R.Components.Plots.Implementation.Commands {
+    internal sealed class PlotDevicePasteCommand : PlotDeviceCommand, IAsyncCommand {
+        public PlotDevicePasteCommand(IRInteractiveWorkflow interactiveWorkflow, IRPlotDeviceViewModel viewModel)
+            : base(interactiveWorkflow, viewModel) {
+        }
+
+        public CommandStatus Status {
+            get {
+                if (!IsInLocatorMode && Clipboard.GetDataObject().GetDataPresent(PlotClipboardData.Format)) {
+                    return CommandStatus.SupportedAndEnabled;
+                }
+
+                return CommandStatus.Supported;
+            }
+        }
+
+        public async Task<CommandResult> InvokeAsync() {
+            if (Clipboard.ContainsData(PlotClipboardData.Format)) {
+                var source = PlotClipboardData.Parse((string)Clipboard.GetData(PlotClipboardData.Format));
+                if (source != null) {
+                    if (source.ProcessId == InteractiveWorkflow.RSession.ProcessId) {
```

Comment:
I don't know yet.  If we do get a process id, then it should work.  If process id is null, then it will be equal and so will proceed and fail with an error like "source device not found" instead of the more specific error I have here.